src/base1/test-chan.js

```diff
@@ -33,25 +33,28 @@ function MockPeer() {
 
         command = JSON.parse(payload);
         if (command.command == "open") {
+            let reply;
             if (command.payload == "echo") {
                 echos[command.channel] = true;
-                self.send(null, JSON.stringify({
+                reply = {
                     command: "ready",
                     channel: command.channel
-                }));
+                };
             } else if (command.payload == "null") {
                 nulls[command.channel] = true;
-                self.send(null, JSON.stringify({
+                reply = {
                     command: "ready",
                     channel: command.channel
-                }));
+                };
             } else {
-                self.send(null, JSON.stringify({
+                reply = {
                     command: "close",
                     channel: command.channel,
                     problem: "not-supported",
-                }));
+                };
             }
+            // HACK: send the reply asynchronously, so that mock_peer addEventListener() gets to see it before our own onrecv()
```

Comment:
I wouldn't call this a HACK.

Reply:
Heh, indeed. It's more like a weird effect of what the Mock* stuff in the test does.